src/app/learnocaml_partition_view.ml

```diff
@@ -80,7 +89,36 @@ let render_classes xs =
 
 let sum_with f = List.fold_left (fun acc x -> acc + f x) 0
 
-let exercises_tab part =
+let rec students_partition students part =
+  let open Student in
+  match part with
+    |[] -> []
+    |t::q -> let student_t = List.find (fun student -> student.token = t) students in
+             student_t :: students_partition students q
+
+let list_of_students_details students part=
```

Comment:
Are you fluent w.r.t. fold_left and fold_right?

if yes, you can do this here as a follow-up of the previous comment:

```suggestion
let list_of_students_details students part =
  let students =
      List.fold_left (fun res st -> Token.Map.add st.Student.token st res)
        Token.Map.empty students in
```

(if no, we'll talk about this ^^)